blueflood-http/src/main/java/com/rackspacecloud/blueflood/tracker/Tracker.java

```diff
@@ -0,0 +1,196 @@
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.tracker;
+
+import com.rackspacecloud.blueflood.http.HTTPRequestWithDecodedQueryParams;
+import com.rackspacecloud.blueflood.io.Constants;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.management.JMX;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+import java.lang.management.ManagementFactory;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class Tracker implements TrackerMBean {
+
+    private static final Logger log = LoggerFactory.getLogger(Tracker.class);
+    private Set tenantIds = new HashSet();
+
+    private static final String trackerName = String.format("com.rackspacecloud.blueflood.tracker:type=%s", Tracker.class.getSimpleName());
+    static MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
+
+    public Tracker() {
+        registerMBean();
+    }
+
+    public void addTenant(String tenantId) {
+        if (!this.tenantIds.contains(tenantId)) {
+            this.tenantIds.add(tenantId);
+        }
+        log.info("[TRACKER] tenantId " + tenantId + " added.");
+    }
+
+    public void removeTenant(String tenantId) {
+        this.tenantIds.remove(tenantId);
+        log.info("[TRACKER] tenantId " + tenantId + " removed.");
+    }
+
+    public void removeAllTenants() {
+        this.tenantIds.clear();
+        log.info("[TRACKER] all tenants removed.");
+    }
+
+    public boolean isTracking(String tenantId) {
+        return this.tenantIds.contains(tenantId);
+    }
+
+    public Set getTenants() {
+        return this.tenantIds;
+    }
+
+    public void registerMBean() {
+        try {
+            ObjectName objectName = new ObjectName(trackerName);
+
+            // register TrackerMBean only if not already registered
+            if (!mBeanServer.isRegistered(objectName)) {
+                ManagementFactory.getPlatformMBeanServer().registerMBean(this, objectName);
+            }
+        } catch (Exception exc) {
+            log.error("Unable to register mbean for " + this.getClass().getSimpleName(), exc);
+        }
+    }
+
+    public static void track(HttpRequest request) {
+        // check if tenantId is being tracked by JMX TenantTrackerMBean and log the request if it is
+        if (request == null) return;
+        if (mBeanServer == null) return;
+
+        try {
+            ObjectName objectName = new ObjectName(trackerName);
+            if (mBeanServer.isRegistered(objectName)) {
+                // log if tenantId is being tracked
+                TrackerMBean tracker = JMX.newMBeanProxy(mBeanServer, objectName, TrackerMBean.class);
```

Comment:
Ok, we can go this route but we can't not have a static mbean operation so we will have to remove isTracking() from the MBean interface, which is ok.  We can use getTenants() to check instead.